internal/eventstore/internal/repository/sql/push.go

```diff
@@ -0,0 +1,117 @@
+package sql
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/caos/logging"
+	"github.com/caos/zitadel/internal/errors"
+	"github.com/caos/zitadel/internal/eventstore/models"
+	"github.com/cockroachdb/cockroach-go/crdb"
+)
+
+const insertStmt = "insert into eventstore.events " +
+	"(event_type, aggregate_type, aggregate_id, aggregate_version, creation_date, event_data, modifier_user, modifier_service, modifier_tenant, resource_owner, previous_sequence) " +
+	"select $1, $2, $3, $4, coalesce($5, now()), $6, $7, $8, $9, $10, " +
+	// case is to set the highest sequence or NULL in previous_sequence
+	"case (select exists(select event_sequence from eventstore.events where aggregate_type = $11 AND aggregate_id = $12)) " +
+	"WHEN true then (select event_sequence from eventstore.events where aggregate_type = $13 AND aggregate_id = $14 order by event_sequence desc limit 1) " +
+	"ELSE NULL " +
+	"end " +
+	"where (" +
+	// exactly one event of requested aggregate must have a >= sequence (last inserted event)
+	"(select count(id) from eventstore.events where event_sequence >= $15 AND aggregate_type = $16 AND aggregate_id = $17) = 1 OR " +
+	// previous sequence = 0, no events must exist for the requested aggregate
+	"((select count(id) from eventstore.events where aggregate_type = $18 and aggregate_id = $19) = 0 AND $20 = 0)) " +
+	"RETURNING id, event_sequence, creation_date"
+
+func (db *SQL) PushEvents(ctx context.Context, aggregates [][]*models.Event) (err error) {
```

Comment:
rensme aggregates to events, and there are double []

Reply:
PushEvents is deleted